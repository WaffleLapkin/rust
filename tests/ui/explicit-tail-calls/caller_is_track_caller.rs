#![expect(incomplete_features)]
#![feature(explicit_tail_calls)]

#[track_caller]
fn a() {
    become b();
    //~^ error: a function marked with `#[track_caller]` cannot perform a tail-call
}

fn b() {}

#[track_caller]
fn c() {
    become a();
    //~^ error: a function marked with `#[track_caller]` cannot perform a tail-call
    //~| error: a function marked with `#[track_caller]` cannot be tail-called
}

fn main() {}
