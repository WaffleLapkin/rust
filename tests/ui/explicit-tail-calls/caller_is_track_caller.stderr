error: a function marked with `#[track_caller]` cannot perform a tail-call
  --> $DIR/caller_is_track_caller.rs:6:5
   |
LL |     become b();
   |     ^^^^^^^^^^

error: a function marked with `#[track_caller]` cannot perform a tail-call
  --> $DIR/caller_is_track_caller.rs:14:5
   |
LL |     become a();
   |     ^^^^^^^^^^

error: a function marked with `#[track_caller]` cannot be tail-called
  --> $DIR/caller_is_track_caller.rs:14:5
   |
LL |     become a();
   |     ^^^^^^^^^^

error: aborting due to 3 previous errors

